packages/ra-core/src/dataProvider/useDataProvider.ts

```diff
@@ -188,52 +184,13 @@ const useDataProvider = <
                         logoutIfAccessDenied,
                         allArguments,
                     };
-                    if (isOptimistic) {
```

Comment:
I had to remove that piece of logic, and it breaks undoable mode because read queries now execute immediately after an optimistic mutation and ovewrite the local mutation. 

But we had no tests for that, and I'm not yet sure how to do undoable mode with react-query. One think is sure: it won't be in useDataProvider.